### ISSUE TITLE
[3.3.5] Core/Spell: WarlockWarlock pet succubus charm spells

### ISSUE THREAD

**Issue by w5860363** [TrinityCore/TrinityCore#16628]
core:https://github.com/TrinityCore/TrinityCore/commit/655fc0135fe5d320cbcea766242468185b3c586a
DB:TDB602016_02_15_00_world.sql
update:
Summon Succubus （http://www.wowhead.com/spell=712）

Summon biological seduction spell to not fight in the cast are sure to fail, even if the Warlock already has in the battle, until the succubus using a normal attack

Please excuse my English.

**Comment by c9240709-b07d-4439-9212-94f05f7a3212** [comment]
Maybe it will be fixed by https://github.com/TrinityCore/TrinityCore/pull/14813/.

**Comment by w5860363** [comment]
@c9240709-b07d-4439-9212-94f05f7a3212 Work but not immediately effective, the need to reduce the level of 60 to 70 and then upgrade to

**Comment by shenhuyong** [comment]
Confirmed.
This bug has been in existence since rev. 233297c5c8619af96e2aab4fa5ef6d205b5dc18a
And I fixed it.

Find these code in Spell.cpp
`    // if the spell allows the creature to turn while casting, then adjust server-side orientation to face the target now`
`    // client-side orientation is handled by the client itself, as the cast target is targeted due to Creature::FocusTarget`
`    if (m_caster->GetTypeId() == TYPEID_UNIT && !m_caster->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PLAYER_CONTROLLED))`
`        if (!m_spellInfo->HasAttribute(SPELL_ATTR5_DONT_TURN_DURING_CAST))`
`            if (WorldObject* objTarget = m_targets.GetObjectTarget())`
`            {`
`                m_caster->SetInFront(objTarget);`
`                m_caster->SetFacingToObject(objTarget);`
`            }`

Change it to
`    // if the spell allows the creature to turn while casting, then adjust server-side orientation to face the target now`
`    // client-side orientation is handled by the client itself, as the cast target is targeted due to Creature::FocusTarget`
`    if (m_caster->GetTypeId() == TYPEID_UNIT && !m_caster->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PLAYER_CONTROLLED))`
`        if (!m_spellInfo->IsChanneled() && !m_spellInfo->HasAttribute(SPELL_ATTR5_DONT_TURN_DURING_CAST))`
`            if (WorldObject* objTarget = m_targets.GetObjectTarget())`
`            {`
`                m_caster->SetInFront(objTarget);`
`                m_caster->SetFacingToObject(objTarget);`
`            }`

**Comment by Keader** [comment]
Any idea what caused it? (Which commit?)

**Comment by Fiver** [comment]
Maybe is https://github.com/TrinityCore/TrinityCore/commit/f481ae1048aac32ba3a40491f0304d0317234295 caused it!

**Comment by Keader** [comment]
Not cause by f481ae1048aac32ba3a40491f0304d0317234295
I´m revert my TC for rev 7f5805f30841697a67ff37163d08544a3f148414 (commit before f481ae1048aac32ba3a40491f0304d0317234295) and this was already broken

**Comment by sirikfoll** [comment]
I have tried to debug this, as is not an 100% chance to reproduce i'm not sure if this is the right thing, but is the callstack that seems more odd.

Apparently the succubus tried to move when beggining to channel and interrupt it. no idea of why or how to fix
```
    worldserver.exe!Spell::finish(bool ok) Line 3564	C++
 	worldserver.exe!Spell::cancel() Line 3066	C++
 	worldserver.exe!Unit::InterruptSpell(CurrentSpellTypes spellType, bool withDelayed, bool withInstant) Line 3115	C++
 	worldserver.exe!Unit::InterruptNonMeleeSpells(bool withDelayed, unsigned int spell_id, bool withInstant) Line 3175	C++
 	worldserver.exe!Unit::RemoveAurasWithInterruptFlags(unsigned int flag, unsigned int except) Line 4079	C++
 	worldserver.exe!Unit::UpdatePosition(float x, float y, float z, float orientation, bool teleport) Line 17346	C++
 	worldserver.exe!Unit::UpdateSplinePosition() Line 449	C++
 	worldserver.exe!Unit::UpdateSplineMovement(unsigned int t_diff) Line 424	C++
 	worldserver.exe!Unit::Update(unsigned int p_time) Line 391	C++
 	worldserver.exe!Creature::Update(unsigned int diff) Line 570	C++
 	worldserver.exe!Pet::Update(unsigned int diff) Line 641	C++
 	worldserver.exe!Trinity::ObjectUpdater::Visit<Creature>(GridRefManager<Creature> & m) Line 348	C++

**Comment by Keader** [comment]
Already fixed by 66006e3d0dd499152ce367e72fbf223356670eb6
i cant reproduce this problem anymore :+1:
